File: pset6/caesar.py
from sys import argv
from cs50 import get_string

if len(argv) == 2:
    key = int(argv[1])

    if key == 0:
        print("Usage: python caesar.py key")
        exit(1)

    else:
        s = get_string("plaintext: ")
        print("ciphertext: ", end="")

        for i in range(len(s)):
            c = ord(s[i])

            if c >= 97 and c <= 122 or c >= 65 and c <= 90:

                if c >= 97 and c <= 122:
                    c += key

                    while c >= 123:
                        c -= 122
                        c += 96

                if c >= 65 and c <= 90:
                    c += key

                    while c >= 91:
                        c -= 90
                        c += 64

                a = chr(c)
                print(a, end="")

            else:
                b = chr(c)
                print(b, end="")

        i += 1
    print()

if len(argv) != 2:
    print("Usage: python caesar.py key")
    exit(1)

File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, const char *argv[])
{
    int key;
    int c;
    char a;
//Checking validy of key
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    if (argc == 2)
    {
        key = atoi(argv[1]);
//If in argv[] text or smth, doing this
        if (key == 0)
        {
            printf("Usage: ./caesar key\n");
        }
//Checking once more, if ok, request a text
        if (key >= 1 || key <= 9)
        {
            string s = get_string("plaintext: ");
            printf("ciphertext: ");
//Printing out text and its ascii
            for (int i = 0; i < strlen(s); i++)
            {
                c = (int) s[i];
                if ((c >= 65 && c <= 90) || (c >= 97 && c <= 122))
                {
                    c += key;
                    for (; c >= 123;)
                    {
                        c -= 122;
                        c += 96;
                    }
                    a = (char) c;
                    printf("%c", a);
                }
                else 
                {
                    printf("%c", c);
                }
            }
            printf("\n");
        }
         return 0;
    }
    else 
    {
        printf("Usage: ./caesar key\n"); 
    }
}
